test/types/object.js

```diff
@@ -2163,6 +2164,62 @@ describe('object', () => {
 
     });
 
+    describe('func().class()', () => {
+
+        it('should differentiate between classes and functions', (done) => {
+
+            const classSchema = Joi.object({
+                _class: Joi.func().class()
+            });
+
+            const testFunc = function () {};
+            const testClass = class MyClass {};
+
+            Helper.validate(classSchema, [
+                [{ _class: testClass }, true],
+                [{ _class: testFunc }, false, null, {
+                    message: 'child "_class" fails because ["_class" must be a class]',
+                    details: [{
+                        message: '"_class" must be a class',
+                        path: ['_class'],
+                        type: 'function.class',
+                        context: { key: '_class', label: '_class' }
+                    }]
+                }]
+            ], done);
+        });
+
+        it('refuses class look-alikes and bad values', (done) => {
+
+            const classSchema = Joi.object({
+                _class: Joi.func().class()
+            });
+
+            expect(IsClass('class')).to.equal(false);
+
+            Helper.validate(classSchema, [
+                [{ _class: ['class '] }, false, null, {
+                    message: 'child "_class" fails because ["_class" must be a Function]',
```

Comment:
Interesting, so I'm guessing this is the side-effect of this being off `Joi.func()` where we have the `this._flags.func` flag set for validation.

It makes sense given the implementation that we are returned the error `"_class" must be a Function` my initial thought was that we would get `"_class" must be a class`. But returning the base type first is the same way it works for any of our other types (ie. `Joi.string().uuid()`)

Reply:
Yup! So in order to cover the additional function type check, `typeof func === 'function' &&` I had to use the `is-class.js` helper directly in the tests